include/alpaka/mem/view/Accessor.hpp

```diff
@@ -0,0 +1,442 @@
+        {
+            return reinterpret_cast<const char*>(p);
+        }
+    } // namespace internal
+
+    //! Access tag type indicating read-only access.
+    struct ReadAccess
+    {
+    };
+
+    //! Access tag type indicating write-only access.
+    struct WriteAccess
+    {
+    };
+
+    //! Access tag type indicating read-write access.
+    struct ReadWriteAccess
+    {
+    };
+
+    //! An accessor is an abstraction for accessing memory objects such as views and buffers.
+    //! @tparam TMemoryHandle A handle to a memory object.
+    //! @tparam TElem The type of the element stored by the memory object. Values and references to this type are
+    //! returned on access.
+    //! @tparam TBufferIdx The integral type used for indexing and index computations.
+    //! @tparam TDim The dimensionality of the accessed data.
+    //! @tparam TAccessModes Either a single access tag type or a `std::tuple` containing multiple access tag types.
+    template<typename TMemoryHandle, typename TElem, typename TBufferIdx, std::size_t TDim, typename TAccessModes>
+    struct Accessor;
+
+    namespace internal
+    {
+        template<typename T>
+        struct WriteOnlyProxy
+        {
+            ALPAKA_FN_HOST_ACC WriteOnlyProxy(T& location) : loc(location)
+            {
+            }
+
+            template<typename U>
+            ALPAKA_FN_HOST_ACC auto& operator=(U&& value)
+            {
+                loc = std::forward<U>(value);
+                return *this;
+            }
+
+        private:
+            T& loc;
+        };
+
+        template<typename TPointer, typename TElem, typename TAccessModes>
+        struct AccessReturnTypeImpl;
+
+        template<typename TPointer, typename TElem>
+        struct AccessReturnTypeImpl<TPointer, TElem, ReadAccess>
+        {
+            using type = TElem;
+        };
+
+        template<typename TPointer, typename TElem>
+        struct AccessReturnTypeImpl<TPointer, TElem, WriteAccess>
+        {
+            using type = WriteOnlyProxy<TElem>;
+        };
+
+        template<typename TPointer, typename TElem>
+        struct AccessReturnTypeImpl<TPointer, TElem, ReadWriteAccess>
+        {
+            using type = TElem&;
+        };
+
+        template<typename TPointer, typename TElem, typename THeadAccessMode, typename... TTailAccessModes>
+        struct AccessReturnTypeImpl<TPointer, TElem, std::tuple<THeadAccessMode, TTailAccessModes...>>
+            : AccessReturnTypeImpl<TPointer, TElem, THeadAccessMode>
+        {
+        };
+
+        template<typename TPointer, typename TElem, typename TAccessModes>
+        using AccessReturnType = typename internal::AccessReturnTypeImpl<TPointer, TElem, TAccessModes>::type;
+    } // namespace internal
+
+    //! 1D accessor to memory objects represented by a pointer.
+    template<typename TPointer, typename TElem, typename TBufferIdx, typename TAccessModes>
+    struct Accessor<TPointer, TElem, TBufferIdx, 1, TAccessModes>
```

Comment:
Shouldn't there be a `std::is_pointer_v` somewhere around here? Otherwise `TPointer` will match a lot more than pointers.

Reply:
I inteded `TPointer` to not only be a `T*` but more kinds of pointers as well (like smart pointers, fat pointers, fancy pointers, etc.). But I probably need to be more specific what kind of pointers fit here.

Reply:
I dropped support for custom pointers. View accessors expect raw pointers as memory handles now.